R/mda_helpers.R

```diff
@@ -1,31 +1,51 @@
-#'@title create timeseries pointer
-#'@param variable a timeseries name (e.g., \code{wtr})
-#'@return a timseries string
-#'@export
-make_ts_variable <- function(variable){
-  ts_prefix <- get_ts_prefix()
-  ts_variable = paste0(ts_prefix, variable)
-  return(ts_variable)
+#' Translate a timeseries name from mda.streams to ScienceBase
+#' @param variable a timeseries name[s] in mda.streams lingo (e.g., \code{wtr})
+#' @return timseries name[s] in ScienceBase lingo (e.g., "\code{ts_wtr})
+make_ts_name <- function(variable){
+  paste0(pkg.env$ts_prefix, variable)
 }
 
-get_ts_prefix <- function(){
-  ts_prefix = 'ts_'
-  return(ts_prefix)
+#' Translate a timeseries name from ScienceBase to mda.streams
+#' @param ts_name timeseries name[s] in ScienceBase lingo (e.g., "\code{ts_wtr})
+#' @param use_names logical. Should the return vector be named according to the input values?
+#' @return timeseries name[s]in mda.streams lingo (e.g., \code{wtr})
+parse_ts_name <- function(ts_name, use_names=TRUE) {
+  vapply(strsplit(ts_name, '_'), '[', vector('character', length = 1), 2, USE.NAMES=use_names)
 }
 
-get_ts_extension <- function(){
-  ts_extension = 'tsv'
-  return(ts_extension)
+#' Translate a site name from database and site number to a 
+#' ScienceBase+mda.streams site ID
+#' @param database a character or character vector of databases, probably in 
+#'   \code{c("nwis","nldas")}.
```

Comment:
I think we currently only do nwis, although we can support other site types ("bobs" or other groups?). nldas is a dataset that we sample at these site locations, so it probably doesn't fit here.